config/v1/types_image_content_source_policy.go

```diff
@@ -0,0 +1,43 @@
+package v1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ImageContentSourcePolicy holds cluster-wide information about how to handle registry mirror rules.  The canonical name is `cluster`
+type ImageContentSourcePolicy struct {
+	metav1.TypeMeta `json:",inline"`
+	// Standard object's metadata.
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +required
+	Spec ImageContentSourcePolicySpec `json:"spec"`
+}
+
+type ImageContentSourcePolicySpec struct {
+	// DigestMappingRules holds the registry mirroring rules that are written to
```

Comment:
Do we want to document this as a general-purpose facility, or specifically restricted to OpenShift product images?  It works just the same for both for unauthenticated registries, but we can only supply mirror credentials via a CRI-O-wide option (not via the CRI).  Or maybe we don’t need to restrict this to OpenShift only, but say that non-OpenShift repositories that require credentials are {problematic,not recommended,not supported}.